packages/go_router/lib/src/route_data.dart

```diff
@@ -47,7 +47,7 @@ abstract class GoRouteData {
   /// Subclasses must override one of [build], [buildPage], or [redirect].
   ///
   /// Corresponds to [GoRoute.redirect].
-  FutureOr<String?> redirect() => null;
+  FutureOr<String?> redirect(BuildContext context, GoRouterState state) => null;
```

Comment:
This will be a hard break for go_router_builder, we may need to produce a new API to migrate go_router_builder